src/redux-middleware/pullBeforeMove.ts

```diff
@@ -0,0 +1,35 @@
+import { ThunkMiddleware } from 'redux-thunk'
+import pull from '../action-creators/pull'
+import { isPending, pathExists, simplifyPath } from '../selectors'
+import { hashContext, head, parentOf, pathToContext } from '../util'
+import { State } from '../util/initialState'
+
+/**
+ * Middleware that fetches any pending descendants before a thought-move operation.
+ */
+const pullBeforeMove: ThunkMiddleware<State> = ({ getState, dispatch }) => {
+
+  return next => async action => {
```

Comment:
Could you explain how asynchronous middleware works? What happens to other actions that have been dispatched while this is fetching?

Reply:
It doesn't block the other actions that are dispatched while it's fetching